packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -48,6 +48,59 @@ export default util.createRule({
   create(context) {
     const sourceCode = context.getSourceCode();
     return {
+      'LogicalExpression[operator=||],LogicalExpression[operator=??]'(
```

Comment:
This _seems_ right to me, but the rule elsewhere has a bunch of checks on the kinds of binary expressions in play so I'd want to page in @bradzacher for a double check before I merge this.

Reply:
@JoshuaKGoldberg @bradzacher Any updates?

Reply:
I also think [DeMorgan’s Laws](https://ntgard.medium.com/https-medium-com-ntgard-boolean-logic-in-javascript-part-2-ec0b13169b2#:~:text=!(A%20%7C%7C%20B)%20%3D%3D%3D%20!A%20%26%26%20!B) can also be applied to the rule to cover more cases.
`!foo || !foo.bar` (no fix suggest) is equivalent to `!(foo && foo.bar)` (suggests `!(foo?.bar)`)
Should I open an issue for it?

Reply:
(nitpick on styling)
```suggestion
      'LogicalExpression[operator="||"], LogicalExpression[operator="??"]'(
```